projects/observability/src/shared/components/topology/d3/layouts/graph-layout.ts

```diff
@@ -1,76 +1,75 @@
-import { RenderableTopology, RenderableTopologyNode, TopologyCoordinates, TopologyEdge, TopologyNode } from "../../topology";
+import { RenderableTopology, RenderableTopologyNode, TopologyEdge, TopologyNode } from "../../topology";
 
 export class GraphLayout {
 
-  private levelToNodesMap = new Map<number, RenderableTopologyNode[]>();
+  private levelToNodesMap = new Map<number, RenderableTopologyNode[]>([[0, []]]);
   private nodeToLevelMap = new Map<RenderableTopologyNode, number>();
-  private nodeCoOrdinatesMap: Map<TopologyNode, TopologyCoordinates> = new Map<TopologyNode, TopologyCoordinates>();
 
   public layout(topology: RenderableTopology<TopologyNode, TopologyEdge>): void {
 
     this.findRootNodes(topology);
     this.fillNodeAndLevelMaps();
-    this.fillNodeCoOrdinatesMap();
-    this.setNodeCoOrdinates(topology);
+    this.assignCoordinatesToNodes();
   }
 
   private findRootNodes(topology: RenderableTopology<TopologyNode, TopologyEdge>) {
     topology.nodes.forEach(node => {
       if (node.incoming.length === 0) {
         this.nodeToLevelMap.set(node, 0);
-        this.levelToNodesMap.has(0) ? this.levelToNodesMap.get(0)?.push(node) : this.levelToNodesMap.set(0, [node])
+        this.levelToNodesMap.get(0)?.push(node);
       }
     });
   }
 
   private fillNodeAndLevelMaps() {
 
-    const nodes = [...(this.levelToNodesMap.get(0) ?? [])];
-    const visited = new Set<TopologyNode>([...nodes.map(node => node.userNode)]);
+    const nodesThatAreGoingToBeExplored = [...(this.levelToNodesMap.get(0) ?? [])];
+    const nodesThatHaveBeenOrAreGoingToBeExplored = new Set<TopologyNode>([...nodesThatAreGoingToBeExplored.map(node => node.userNode)]);
```

Comment:
nit - you call them both nodes, but you switch between working with `RenderableTopologyNode` and `TopologyNode`. Doesn't seem to be a clear reason why the distinction - we're just doing more work each time we work with this to map to userNode (lines 27,46,47). Only reason would be if there's some possibility that the same userNode could show up in multiple different renderable nodes, but don't believe that's the case.

Reply:
Oh that `.map` was supposed to be removed.
Updated